x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go

```diff
@@ -303,6 +336,10 @@ func createEvents(svc cloudwatchiface.CloudWatchAPI, listMetricsTotal []cloudwat
 				if len(labels) == 4 {
 					identifierValue := labels[identifierValueIdx]
 					events[identifierValue] = insertMetricSetFields(events[identifierValue], output.Values[timestampIdx], labels)
+					tags := resourceTagMap[identifierValue]
+					for _, tag := range tags {
+						events[identifierValue].ModuleFields.Put("tags."+*tag.Key, *tag.Value)
```

Comment:
This more looks like `labels` in ECS then tags. The question is now if we want naming wise be consistent with ECS or AWS. It's under the AWS prefix but I would prefer using the ECS naming even inside if possible.

Reply:
Good point, thanks @ruflin. If I remember correctly, AWS and Azure uses `tags` and GCP uses `labels`. I checked ECS documentation https://www.elastic.co/guide/en/ecs/current/ecs-base.html, it also has `tags` but it's a list of strings instead of key value pairs. 

The only concern I have is, AWS uses `tags` everywhere and if we switch to `labels` here, it will create inconsistency.

Reply:
I'd prefer to use AWS naming (`aws.tags`), specially if we keep it under the `aws` namespace. Though I understand the benefits of using the ECS field. Would it make sense to have a compromise solution by putting the tags on the ECS `labels`, but also configuring `aws.tags` as an alias to `labels`?

Reply:
I think it makes sense to use alias here!

Reply:
I don't think we can use alias here as it would have to alias a dynamic number of fields.

One thing we discussed a few times in context of ECS is having generic templates for example for an `*.hostname` field that it is always a keyword. Applying this to `labels` would mean, that `*.labels` is always an object and a list of `key -> value` pairs. So the above would break this.

My proposal would be to go with `aws.labels` because of this. As the labels we think of in ECS are normally manually defined labels, I wonder if the aws ones fit into the top level?

The main problem I see is that users familiar with aws will search for the `tags` but can't find them :-(

Reply:
> My proposal would be to go with `aws.labels` because of this. As the labels we think of in ECS are normally manually defined labels, I wonder if the aws ones fit into the top level?

The problem with the top-level `labels` is that it could mix labels with different meanings. For example an event originated from an application running on Kubernetes on AWS could be enriched with kubernetes labels and annotations, docker labels and aws tags, the interpretation of all these labels could be different even if they are kind of similar. So I think we are always going to need different names for these containers of labels. And given that, I think that I'd prefer to use the name the users are familiar with. I think it is ok to need an additional mapping.

(Maybe another option thinking on ECS would be to have all the labels under a two level `labels.*.*`, so we have:  `labels.docker`, `labels.kubernetes`, `labels.kubernetes_annotations`, `labels.aws_tags` and so on. Something to discuss out of this PR :slightly_smiling_face: )

Reply:
This conversation comes up every time we add a service that has labels/tags. I suggest we move forward with `aws.tags` until we find a global solution (if any) :+1: